scilpy/segment/streamlines.py

```diff
@@ -2,56 +2,48 @@
 
 import itertools
 
+from dipy.io.stateful_tractogram import StatefulTractogram
 from dipy.tracking.metrics import length
-from dipy.tracking.streamline import (set_number_of_points,
-                                      transform_streamlines)
-from dipy.tracking._utils import _mapping_to_voxel
+from dipy.tracking.streamline import set_number_of_points
 from dipy.tracking.vox2track import _streamlines_in_mask
 from nibabel.affines import apply_affine
 import numpy as np
 
 
-def target_line_based(streamlines, target_mask, affine=None, include=True):
+def target_line_based(sft, target_mask):
+    sft.to_vox()
+    sft.to_corner()
     # Copy-Paste from Dipy to get indices
+    # Should always be in vox/center space
     target_mask = np.array(target_mask, dtype=np.uint8, copy=True)
-    lin_T, offset = _mapping_to_voxel(affine)
 
-    streamline_index = _streamlines_in_mask(
-        streamlines, target_mask, lin_T, offset)
+    streamline_index = _streamlines_in_mask(list(sft.streamlines), target_mask,
+                                            np.eye(3), [-0.5, -0.5, -0.5])
 
```

Comment:
I don't really like the name... First, shouldn't it be plural streamlines_indices? It's not really an index, nor a list of indices. It's 0,1 or 2. streamlines_labels? in_mask_bools? I don't know.

Reply:
It is a function from dipy, the dipy function is pretty crappy I agree. I
I will remove the return of streamlines. We only need the indices, which will also make this script easier to understand.
The indices are needed because of the sequential filtering. I will update the description and change the function name (and the variable)